support condition branch in ao debug handler

Gasoonjia opened the pull request:
Summary:
This diff introduced the supportive of condition statement into ao debug handler generation.

Most of code borrowed from ExecuTorch to avoid circle dependency issue.

Test Plan: buck2 test fbcode//caffe2/test/quantization:quantization -- test_control_flow

Reviewed By: jerryzh168

Differential Revision: D65616927

jerryzh168 reviewed (changes requested):
we don't want to change the signature for `generate_numeric_debug_handle` I think

jerryzh168 commented:
can you update the summary to talk about the discussion of GraphModule v.s. ExportedProgram as well

Gasoonjia committed:
support condition branch in ao debug handler (#140256)

Summary:

This diff introduced the supportive of condition statement into ao debug handler generation.

Most of code borrowed from ExecuTorch to avoid circle dependency issue.

Test Plan: buck2 test fbcode//caffe2/test/quantization:quantization -- test_control_flow

Reviewed By: jerryzh168

Differential Revision: D65616927
